### ISSUE TITLE
[fix] On branch checkout I cannot change the local changes behavior

### ISSUE THREAD

**Issue by albertodlc** [sourcegit-scm/sourcegit#748]
I changed the checkout behavior from `local changes: Do nothing` to `local changes: discard` and now I don't see how can change it back. When I press checkout, the popup does not allow to change the radiobutton. It automatically discard them.

I think the way this is handled should be changed, the popup should always display and allow the user to change and press the confirm button. In my case I changed the behavior for a specific checkout, and I lost some changes later because I think the checkout would ask for confirmation.

![image](https://github.com/user-attachments/assets/f08626f3-85bd-4aab-aaaa-e695d44208af)

**Comment by albertodlc** [comment]
By  checkout behavior and local changes radiobuttons I mean this popup:
![image](https://github.com/user-attachments/assets/170b66c2-d61a-4999-9173-f3e1dc839567)

**Comment by love-linger** [comment]
I can NOT reproduce this issue.

https://github.com/user-attachments/assets/7a1db55a-faff-4f6c-b8bd-81bf1b4a4ecd


If there are some un-committed local changes, the `Checkout` popup should always show. Try to press and release `Ctrl` (`Command` on macOS) again - It seems it triggred the `Hold Ctrl to start directly`

**Comment by love-linger** [comment]
You can try the latest build from Github Action.

**Comment by love-linger** [comment]
P.S. this is why I do NOT like to add the feature to remember the last user selection.

**Comment by albertodlc** [comment]
Ok thank you, I reinstalled the last public build and worked fine again, so I cannot test the fix with Github's actions last build, but thanks for the quick response. 

I tried to replicate the error but I was not capable. As extra info, the only strange thing that I did before the checkout error was to `stage` the changes (without committing).

0. Previously mark the option Discard on a previous checkout
1. On the new branch. Make some changes. Stage those changes
2. Go to Histories and checkout to the new branch (I am not sure, but I think in this moment the popup did not ask for confirmation and executed the checkout with the previous settings - Discard)

I agree with you about this https://github.com/sourcegit-scm/sourcegit/issues/748#issuecomment-2503516892
